### PR TITLE
fix: fix a few TS errors

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

The TypeScript compiler screams at me again after bumping to `beta.5` when I ran `tsc` in my site repo. It turns out that CI checks fall short when it comes to the `.d.ts` files, so these mistakes tend to sneak in unnoticed.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test plan

```
cd website && yarn typecheck
```

This actually works well. Should we add it to the CI?

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Okay this is annoying😅 As the `website` is also built into a PWA, it uses the `workbox` library which requires setting TS compiler to `lib: ["webworker"]`, but `"webworker"` and `"DOM"` are incompatible. Therefore running `typecheck` in `website` is sure to fail

Not sure what we should do

File: packages/docusaurus-init/templates/classic-typescript/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "compilerOptions": {
     "paths": {
       "@site/*": ["./*"]
-    }
+    },
+    "baseUrl": "."
   },
   "include": ["src/"]
```
Comment (slorber) [inline packages/docusaurus-init/templates/classic-typescript/tsconfig.json:10]: are baseUrl + include really necessary? baseUrl is in base tsconfig and include has been removed by orta

also `@site/` alias can be removed now?

File: website/tsconfig.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "paths": {
       "@site/*": ["./*"]
     },
+    "baseUrl": ".",
     "resolveJsonModule": true
   },
   "include": ["src/"]
```
Comment (slorber) [inline website/tsconfig.json:11]: same comments as for the template

**Review by slorber — COMMENTED** [review]
> Okay this is annoying😅 As the website is also built into a PWA, it uses the workbox library which requires setting TS compiler to lib: ["webworker"], but "webworker" and "DOM" are incompatible. Therefore running typecheck in website is sure to fail. Not sure what we should do

This seems to fix it:
- `"exclude": ["src/sw.js"]`
- Also `skipLibChecks: true` but the 1st is better. I don't think it's too important to type-check this file 🤪 

> Also, the new tsconfig still doesn't do its magic on resolving @site/ aliases


Weird 🤪 I can't make it work either

I guess it's fine to merge this for now and figure out why the site alias can't be in parent config

File: packages/docusaurus-init/templates/classic-typescript/tsconfig.json (hunk 1 above)
Comment (Josh-Cena) [reply packages/docusaurus-init/templates/classic-typescript/tsconfig.json:10]: `baseUrl` is actually required; `include` I don't know. But putting it there makes me feel safe as other files won't be accidentally included

`@site/` is used in the `versions.tsx` page and removing it causes errors

**Comment by slorber** [comment]
Added `yarn workspace website tsc` to our test workflow, it should help catch those `d.ts` issues

Thanks 👍

File: packages/docusaurus-init/templates/classic-typescript/tsconfig.json (hunk 1 above)
Comment (slorber) [reply packages/docusaurus-init/templates/classic-typescript/tsconfig.json]: hmmm it seems I got it working by keeping just `baseUrl`

**Comment by Josh-Cena** [comment]
Okay as long as `tsc` passes I'm fine 😏

**Pull request merged**